### PR TITLE
HDX-9847 Update the food_security endpoint

### PR DESCRIPTION
**Pull request by IanHopkinson**
This PR updates the food_security endpoint to match the new schema.

The tests pass and the endpoint looks ok in `docs`.

I'm a bit concerned I haven't handled the reference start/end/min/max situation correctly

### CODE REVIEW

File: hdx_hapi/endpoints/get_food_security.py [hunk 1]
```diff
@@ -31,77 +32,51 @@
 
 
 @router.get(
-    '/api/themes/food_security',
+    '/api/food/food_security',
```
Comment (danmihaila) [inline hdx_hapi/endpoints/get_food_security.py]: it has to be 'food-security'

File: hdx_hapi/endpoints/get_food_security.py [hunk 2]
```diff
@@ -31,77 +32,51 @@
...
     response_model=HapiGenericResponse[FoodSecurityResponse],
     summary='Get food security data',
     include_in_schema=False,
 )
 @router.get(
-    '/api/v1/themes/food_security',
+    '/api/v1/food/food_security',
```
Comment (danmihaila) [inline hdx_hapi/endpoints/get_food_security.py]: same, food-security

File: tests/test_endpoints/test_food_security_endpoint.py [hunk 3]
```diff
@@ -8,7 +8,7 @@
 
 log = logging.getLogger(__name__)
 
-ENDPOINT_ROUTER = '/api/themes/food_security'
+ENDPOINT_ROUTER = '/api/v1/food/food_security'
```
Comment (danmihaila) [inline tests/test_endpoints/test_food_security_endpoint.py]: same as above, food-security

File: tests/test_endpoints/endpoint_data.py [hunk 4]
```diff
@@ -229,34 +229,31 @@
             'admin2_name',
         ],
     },
-    '/api/themes/food_security': {
+    '/api/v1/food/food_security': {
```
Comment (danmihaila) [inline tests/test_endpoints/endpoint_data.py]: food-security

Comment (IanHopkinson) [reply tests/test_endpoints/endpoint_data.py]: At least I'm consistent! I'll fix these

File: hdx_hapi/db/models/views/all_views.py [hunk 5]
```diff
@@ -176,21 +210,20 @@ class HumanitarianNeedsView(Base):
...
     reference_period_start: Mapped[DateTime] = column_property(location_view.c.reference_period_start)
     reference_period_end: Mapped[DateTime] = column_property(location_view.c.reference_period_end)
 
-
 class NationalRiskView(Base):
     __table__ = national_risk_view
     resource_hdx_id: Mapped[str] = column_property(national_risk_view.c.resource_hdx_id)
     location_ref: Mapped[int] = column_property(national_risk_view.c.location_ref)
-    risk_class: Mapped[RiskClass] = column_property(national_risk_view.c.risk_class)
+    risk_class: Mapped[str] = column_property(national_risk_view.c.risk_class)
```
Comment (alexandru-m-g) [inline hdx_hapi/db/models/views/all_views.py]: The type here should be `RiskClass`. The `all_view.py` from the `dev` branch should be the correct one.

File: hdx_hapi/db/models/views/all_views.py [hunk 6]
```diff
@@ -226,26 +259,23 @@ class OperationalPresenceView(Base):
...
     org_type_code: Mapped[str] = column_property(org_view.c.org_type_code)
     org_type_description: Mapped[str] = column_property(org_view.c.org_type_description)
 
-
 class PopulationView(Base):
     __table__ = population_view
     resource_hdx_id: Mapped[str] = column_property(population_view.c.resource_hdx_id)
     admin2_ref: Mapped[int] = column_property(population_view.c.admin2_ref)
-    gender: Mapped[Gender] = column_property(population_view.c.gender)
+    gender: Mapped[str] = column_property(population_view.c.gender)
```
Comment (alexandru-m-g) [inline hdx_hapi/db/models/views/all_views.py]: Similar as above. I think it's easier to just use the `all_views.py` from `dev`

**Review by alexandru-m-g — CHANGES_REQUESTED** [review]

File: tests/test_endpoints/endpoint_data.py (hunk 4 above)
Comment (IanHopkinson) [reply tests/test_endpoints/endpoint_data.py]: These are fixed in beb8895d196c1227d88312de0121d2733fff8678

File: hdx_hapi/db/models/views/all_views.py (hunk 6 above)
Comment (IanHopkinson) [reply hdx_hapi/db/models/views/all_views.py]: These are fixed in c6c4800d307df572c543ef87fe9127cdf7e30920

I don't think my code is able to pick up the enum types, so I just copied over `all_views.py` from `dev` as you suggested.

**Pull request merged**